Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -156,38 +160,104 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
 
   // cursor, with a hax for-loop
   for (unsigned int i = 0; i < 4; ++i)
-    AddInput(new Cursor(!!(i & 2), (&m_state_in.cursor.x)[i / 2], !!(i & 1)));
+    AddInput(new Cursor(!!(i & 2), (&m_current_state.cursor.x)[i / 2], !!(i & 1)));
 
   // Raw relative mouse movement.
   for (unsigned int i = 0; i != mouse_caps.dwAxes; ++i)
   {
-    AddInput(new RelativeMouseAxis(i, false, &m_state_in.relative_mouse));
-    AddInput(new RelativeMouseAxis(i, true, &m_state_in.relative_mouse));
+    AddInput(new RelativeMouseAxis(i, false, &m_current_state.relative_mouse));
+    AddInput(new RelativeMouseAxis(i, true, &m_current_state.relative_mouse));
   }
 }
 
 void KeyboardMouse::UpdateCursorInput()
```

Comment:
This entire function should probably behave like it did before when the mouse jail is disabled. Otherwise you're applying your sensitivity calculations even when it's off, which kinda breaks Wiimote pointing using the mouse.

So basically just do `if (Config::Get(Config::OCTAGONAL_MOUSE_JAIL_ENABLED)) { /* your new mouse jail code */ } else { /* old mouse code */ }`.

Reply:
(Unless you can guarantee that Sensitivity == 1.0 always behaves like it did before, which is not obvious from the code.)

Reply:
Sensitivity = 1.0 _should_ behave like before. Like with any feature, I can't exhaustively test every case, unfortunately. The math behaves the exact same as far as I understand math. 

The sensitivity being enabled without the mouse jail enabled is on purpose. I don't see any reason to remove that functionality for everyone who is not using the mouse jail if they want it.

Reply:
This does not appear to be the case.

Master: https://user-images.githubusercontent.com/4522237/169666764-381983b5-9488-41c1-97e8-86ea145c445d.mp4


This PR with Sensitivity 1: https://user-images.githubusercontent.com/4522237/169666765-1dbbbeeb-879e-4b7f-ace4-056713af9dff.mp4

Reply:
Okay, I have no way of checking that. I can't load into the wii menu with any input method and for some reason the x cursor inputs get deleted when I close the wii menu or melee. If I can figure out a way to test it, I'll fix it.

Reply:
Doesn't have to be the Wii menu, any Wii game with a pointer should do it. (which should be all of them if you hit the Home button)

Reply:
(Also I have no idea what you mean with 'x cursor inputs get deleted')

Reply:
I don't have any wii games. 

If you right-click as if you were assigning cursor X/Y/+/- to a control stick, the inputs in that list are not functional after opening and closing the emulator.  Just a new weird bug

Reply:
Can't say that has happened to me so can't help you with that I'm afraid.

I debugged this, it's twofold: 
- You're feeding the raw_mouse_point (which is relative to the screen, not the window) directly into the calculations. You need to offset it to the window first (that's what the `ScreenToClient()` call did), otherwise your division by the window dimensions isn't meaningful.
- Your aspect ratio adjustment then further skews this. This makes sense for the octagonal gate, since that should be square, but not for emulating a Wiimote pointer. You might need to put that part behind a conditional.

Reply:
:o Thanks! I'll get right on that.

Reply:
Would you mind checking this on the new patch? I think I mostly fixed it. Master was a little wonky by the looks of that video,. I think the only thing that might have changed was the side of the y-axis that got more squished.

Reply:
The 'wonky' you mean is probably because we can't map 1:1 because that's just not how the Wiimote works (we'd have to have separate scale and offset for each game for that! it would be super cool but so much work...). I'll check if it's better now in a bit.

Reply:
Ok, let me know how it goes. And yeah, I think that's what I'm seeing. Just a= bit closer to f100% accurate on one side of the screen than on the other.

Reply:
It's still not the same because you're ignoring the `g_controller_interface.GetWindowInputScale()` value; this adjusts the values to be relative to the actual render part of the window -- ie, removing the black bars around the window from calculation if the window aspect ratio and the game aspect ratio are different.

May I suggest: https://gist.github.com/AdmiralCurtiss/964bf1a17b9ced9e1acf576cceccc9ef
This will produce identical results to before if the jail is off and the sensitivity is 1. Please verify if it still produces correct results for your use-case as well.

That said, this function is... messy, I'm not gonna lie, a lot here could be consolidated and rewritten for readability. Do you mind if I refactor this once we agree on the produced output?

I also question whether the `if (render_window_info.handle == nullptr)` -> `if (jail_enabled)` branch is necessary, does that ever produce anything meaningful?

Reply:
Revision left for whenever you have the time to review again :) I appreciate the feedback.

Reply:
I forgot to say that I would be happy to take any feedback, but I'd prefer you not to directly edit my PR so I have a chance to learn. :)

Reply:
@AdmiralCurtiss Did you get a chance to re-check this when you reviewed this PR?  I'd like to resolve this convo if it's solved for you.

Reply:
Yes, this seems fine now. I'll wait for your next push to resolve it though because of the issue where x is multiplied with 0 when no game is running, since that technically affects this too.

Reply:
Seems good now, yup.